cli/api/dbadapters/bigquery.ts

```diff
@@ -330,72 +330,79 @@ export class BigQueryDbAdapter implements IDbAdapter {
     return retry(
       async () => {
         try {
-          const job = await this.getClient().createQueryJob({
-            useLegacySql: false,
-            jobPrefix: "dataform-" + (jobPrefix ? `${jobPrefix}-` : ""),
-            query,
-            params,
-            labels,
-            location,
-            dryRun
-          });
-          const resultStream = job[0].getQueryResultsStream();
-          return new Promise<IExecutionResult>((resolve, reject) => {
-            if (isCancelled) {
-              resultStream.end();
-              reject(new Error("Query cancelled."));
-              return;
-            }
-            onCancel?.(() => {
-              resultStream.end();
-              reject(new Error("Query cancelled."));
+          if (dryRun && query.includes('-- @dry-run:skip-results-check')) {
+            // If the above evaluated to true, this means we would be running the row count query of an assertion (see execution_sql.ts).
+            // This must be stopped during a dry run, as the view is not actually created. Therefore we skip the creation of a query job
+            // and instead directly return.
+            return { rows: [], metadata: {} };
+          } else {
```

Comment:
This else can be skipped, because the function will already have been returned from the if. Also this will get rid of the indentation that is making reviewing a bit confusing.